libs/cart/state/src/facades/cart/cart.facade.spec.ts

```diff
@@ -1618,4 +1607,100 @@ describe('DaffCartFacade', () => {
       expect(facade.canPlaceOrder$).toBeObservable(expected);
     });
   });
+
+  describe('getCartItemTotalDiscount', () => {
+
+    it('should be the cart item\'s discount total', () => {
+      const cart = cartFactory.create({
+        items: statefulCartItemFactory.createMany(2)
+      });
+      const expected = cold('a', { a: cart.items[0].total_discount });
+      facade.dispatch(new DaffCartLoadSuccess(cart));
+      expect(facade.getCartItemTotalDiscount(cart.items[0].item_id)).toBeObservable(expected);
+    });
+  });
+
+  describe('getCartItemDiscount', () => {
+
+    it('should be the cart item\'s discounted price', () => {
+      const cart = cartFactory.create({
+        items: statefulCartItemFactory.createMany(2)
+      });
+      const expected = cold('a', { a: daffDivide(cart.items[0].total_discount, cart.items[0].qty) });
+      facade.dispatch(new DaffCartLoadSuccess(cart));
+      expect(facade.getCartItemDiscount(cart.items[0].item_id)).toBeObservable(expected);
+    });
+  });
+
+  describe('getCartItemHasDiscount', () => {
+
+    it('should be if the cart item\'s has a discount', () => {
+      const cart = cartFactory.create({
+        items: statefulCartItemFactory.createMany(2)
+      });
+      const expected = cold('a', { a: cart.items[0].total_discount > 0 });
+      facade.dispatch(new DaffCartLoadSuccess(cart));
+      expect(facade.getCartItemHasDiscount(cart.items[0].item_id)).toBeObservable(expected);
+    });
+  });
+
+  describe('getCartItemPrice', () => {
+
+    it('should be the cart item\'s price', () => {
+      const cart = cartFactory.create({
+        items: statefulCartItemFactory.createMany(2)
+      });
+      const expected = cold('a', { a: cart.items[0].price });
+      facade.dispatch(new DaffCartLoadSuccess(cart));
+      expect(facade.getCartItemPrice(cart.items[0].item_id)).toBeObservable(expected);
+    });
+  });
+
+  describe('getCartItemDiscountedPrice', () => {
+
+    it('should be the cart item\'s discounted price', () => {
+      const cart = cartFactory.create({
+        items: statefulCartItemFactory.createMany(2)
+      });
+      const expected = cold('a', { a: daffSubtract(cart.items[0].price, daffDivide(cart.items[0].total_discount, cart.items[0].qty)) });
```

Comment:
I think this should be an additional field on cart-item, so the computation can be done only once in the magento transformer